### PR TITLE
Fixes for running multi cluster E2E tests locally

### LINKED ISSUE #1496: Update dev docs for testing
Which tests do we have, what are they good for, which setup is needed to run them?
Which tests don't run for PR, which ones make sense to run locally?

* e2e tests that require secret tests
* e2e testenv infra setup
* fleet integration tests

* [x] create table which workflow runs which tag
* [x] https://github.com/rancher/fleet/pull/1764
* [x] https://github.com/rancher/fleet/pull/1765

### PR DESCRIPTION
**Pull request by p-se**
Part of #1496

* Makes a change to be able to run the multi-clusters tests locally using the dev scripts and documents the necessary configuration.
* Makes a change to be able to run the multi-clusters tests locally with [nektos/act](https://github.com/nektos/act).

### CODE REVIEW

File: .github/workflows/e2e-multicluster-ci.yml [hunk 1]
```diff
@@ -116,14 +116,25 @@ jobs:
 
           token=$(kubectl get secret -n fleet-default second-token -o go-template='{{index .data "values" | base64decode}}' | yq eval .token -)
           ca=$(kubectl get secret -n cattle-fleet-system fleet-controller-bootstrap-token -o go-template='{{index .data "ca.crt" | base64decode}}')
+          apiServerURL=$(kubectl get node k3d-upstream-server-0 -o jsonpath='{.status.addresses[?(@.type=="InternalIP")].address}')
```
Comment (p-se) [inline .github/workflows/e2e-multicluster-ci.yml:119]: This is a change that was necessary for me as I, if I remember correctly, have extended the range of default networks available to docker, which results in a different network being used at the start. It seems to work well this way.

File: .github/scripts/label-downstream-cluster.sh [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 set -euxo pipefail
 
-ns=${FLEET_E2E_NS-fleet-default}
+ns=${FLEET_E2E_NS_DOWNSTREAM-fleet-default}
```
Comment (p-se) [inline .github/scripts/label-downstream-cluster.sh:5]: I have had issues starting the test environment from the dev scripts _and_ running them with the same values of the those two environment variables:

* FLEET_E2E_NS
* FLEET_E2E_NS_DOWNSTREAM

With this change I've been able to start and run the tests with the same values of those env vars.

File: .github/workflows/e2e-multicluster-ci.yml [hunk 3]
```diff
@@ -116,14 +116,25 @@ jobs:
...
+                --wait \
+              fleet-agent charts/fleet-agent \
+                --set-string labels.env=test \
+                --set apiServerCA="$ca" \
+                --set apiServerURL="https://$apiServerURL:6443" \
+                --set clusterNamespace="fleet-default" \
+                --set token="$token"
+
+          echo "waiting for downstream cluster to be registered..."
+          { grep -q -m 1 "k3d-downstream"; kill $!; } < <(kubectl get cluster -n fleet-default -w)
```
[the remark this answers is not in the document]
Comment (manno) [reply .github/workflows/e2e-multicluster-ci.yml:137]: Yes. `kubectl -w` exits too early? Maybe we need another check to see if fleet cluster resource is ready, too?

File: .github/workflows/e2e-multicluster-ci.yml (hunk 1 above)
Comment (p-se) [reply .github/workflows/e2e-multicluster-ci.yml:119]: I wonder what the difficulties on Mac would be with this line, perhaps I can address them as well?

Comment (manno) [reply .github/workflows/e2e-multicluster-ci.yml:119]: Not sure, maybe it works. Act runs everything in containers? So the address is probably reachable.

Comment (p-se) [reply .github/workflows/e2e-multicluster-ci.yml]: Yes, it runs everything in containers.

**Review by manno — APPROVED** [review]

**Pull request merged**